File: problems/1909_remove_one_element_to_make_the_array_strictly_increasing/solution.py
from typing import List


class Solution:
    def canBeIncreasing(self, nums: List[int]) -> bool:
        def isIncreasing(arr: List[int]) -> bool:
            for i in range(1, len(arr)):
                if arr[i] <= arr[i - 1]:
                    return False
            return True
        for i in range(len(nums)):
            if i > 0 and i < len(nums) - 1 and nums[i - 1] >= nums[i + 1]:
                continue
            if isIncreasing(nums[:i] + nums[i + 1:]):
                return True
        return False

File: problems/1909_remove_one_element_to_make_the_array_strictly_increasing/Solution.java
class Solution {
    public boolean canBeIncreasing(int[] nums) {
        for (int i = 0; i < nums.length; i++) {
            boolean valid = true;
            int prev = -1;
            for (int j = 0; j < nums.length; j++) {
                if (j == i) continue;
                if (nums[j] <= prev) { valid = false; break; }
                prev = nums[j];
            }
            if (valid) return true;
        }
        return false;
    }
}
